### PR TITLE
OCPBUGS-32510: change metrics-server probes for SNO

### PR DESCRIPTION
**Pull request by simonpasquier**
This change switches the metrics-server's readiness probe to use the `/livez` endpoint instead of `/readyz` for single-node deployments.

By default, the `/readyz` endpoint is used to assert the component readiness. This endpoint returns success when the metrics-server has metric samples over 2 intervals (e.g. it has scraped at least one kubelet twice).

In single-node deployments, it happens sometimes (especially in end-to-end tests) that the kubelet fails to respond in a timely fashion due to contention in cAdvisor, leading to a delayed readiness (and test failures). To workaround the issue, we use the `/livez` endpoint in this mode.

The long-term plan is to switch resource metrics from cAdvisor to the CRI stats API (currently an alpha feature). Once it happens, we can remove this change.

* [ ] I added CHANGELOG entry for this change.
* [X] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

**Comment by machine424** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-single-node openshift/api#1865

**Comment by slashpai** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-single-node openshift/api#1865

File: pkg/manifests/manifests_test.go [hunk 1]
```diff
@@ -2856,6 +2856,75 @@ metricsServer:
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "metrics-client-cert",
+			Namespace: "openshift-monitoring",
+		},
+		Data: map[string][]byte{
+			"tls.crt": []byte("bar"),
+			"tls.key": []byte("foo"),
+		},
+	}
+	apiAuthConfigMapData := map[string]string{
```
Comment (juzhao) [inline pkg/manifests/manifests_test.go:2899]: is `requestheader-client-ca-file` needed?

Comment (simonpasquier) [reply pkg/manifests/manifests_test.go:2899]: technically it isn't required by the unit test.

**Comment by slashpai** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-single-node https://github.com/openshift/api/pull/1865

**Comment by machine424** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-single-node openshift/api#1865

**Comment by machine424** [comment]
(even though we already got a green here https://github.com/openshift/cluster-monitoring-operator/pull/2337#issuecomment-2096201375 and no changes were pushed later. The test is failing on https://github.com/openshift/cluster-monitoring-operator/pull/2337#issuecomment-2096222403 because of unrelated etcd events)

**Comment by juzhao** [comment]
tested with PR
`launch 4.16.0-0.nightly-2024-05-07-025557,openshift/cluster-monitoring-operator#2337 aws,single-node`
readinessProbe path changed from /readyz to /livez and startupProbe is added
```
$ oc -n openshift-monitoring get pod metrics-server-5cc4cd5f75-5nshz -oyaml
...
    livenessProbe:
      failureThreshold: 3
      httpGet:
        path: /livez
        port: https
        scheme: HTTPS
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    name: metrics-server
    ports:
    - containerPort: 10250
      name: https
      protocol: TCP
    readinessProbe:
      failureThreshold: 6
      httpGet:
        path: /livez
        port: https
        scheme: HTTPS
      initialDelaySeconds: 20
      periodSeconds: 20
      successThreshold: 1
      timeoutSeconds: 1
    resources:
      requests:
        cpu: 1m
        memory: 40Mi
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        drop:
        - ALL
      readOnlyRootFilesystem: true
      runAsNonRoot: true
      runAsUser: 1000450000
    startupProbe:
      failureThreshold: 6
      httpGet:
        path: /readyz
        port: https
        scheme: HTTPS
      initialDelaySeconds: 20
      periodSeconds: 20
      successThreshold: 1
      timeoutSeconds: 1
```
/label qe-approved

**Comment by slashpai** [comment]
/payload-job-with-prs periodic-ci-openshift-release-master-nightly-4.16-e2e-aws-ovn-single-node openshift/api#1865

**Pull request merged**